Block/Adminhtml/Form/Field/Indexers.php

```diff
@@ -0,0 +1,119 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Block\Adminhtml\Form\Field;
+
+use Magento\Backend\Block\Template\Context;
+use Magento\Config\Block\System\Config\Form\Field;
+use Magento\Framework\App\Request\Http;
+use Magento\Framework\Data\Form\Element\AbstractElement;
+use Nosto\Tagging\Helper\Scope as NostoHelperScope;
+use Nosto\Tagging\Model\Product\Index\IndexRepository;
+
+class Indexers extends Field
+{
+    /** @var Http $request */
+    public $request;
+
+    /** @var IndexRepository $indexRepository */
+    public $indexRepository;
+
+    /** @var NostoHelperScope $nostoHelperScope */
+    public $nostoHelperScope;
+
+    /**
+     * Indexers block constructor.
+     * @param Context $context
+     * @param Http $request
+     * @param IndexRepository $indexRepository
+     * @param NostoHelperScope $nostoHelperScope
+     * @param array $data
+     */
+    public function __construct(
+        Context $context,
+        Http $request,
+        IndexRepository $indexRepository,
+        NostoHelperScope $nostoHelperScope,
+        array $data = []
+    ) {
+        parent::__construct($context, $data);
+        $this->request = $request;
+        $this->indexRepository = $indexRepository;
+        $this->nostoHelperScope = $nostoHelperScope;
+    }
+
+    /**
+     * Return the amount of products marked as dirty
+     * @return int
+     */
+    public function getAmountDirtyProducts()
+    {
+        $id = (int) $this->request->getParam('store');
+        $store = $this->nostoHelperScope->getStore($id);
+        return $this->indexRepository->getTotalDirty($store);
+    }
+
+    /**
+     * return the amount of products marked as out of sync
+     * @return int
+     */
+    public function getAmountOutOfSyncProducts()
+    {
+        $id = (int) $this->request->getParam('store');
+        $store = $this->nostoHelperScope->getStore($id);
+        return $this->indexRepository->getTotalOutOfSync($store);
+    }
+
+    /**
+     * @param AbstractElement $element
+     *
+     * @return string
+     */
+    protected function _getElementHtml(AbstractElement $element) //@codingStandardsIgnoreLine
+    {
+        return $this->toHtml();
+    }
+
+    /**
+     * @return $this|Field
+     */
+    protected function _prepareLayout() //@codingStandardsIgnoreLine
+    {
+        parent::_prepareLayout();
+        $this->setTemplate('indexersStatuses.phtml');
```

Comment:
Is this correct?

Reply:
Not really, but I'd leave this for later. We would most likely want to do a separate Nosto status page / status view where this should should go.